cmake/Boost_download.cmake.in

```diff
@@ -0,0 +1,82 @@
+cmake_minimum_required(VERSION 3.5)
+include(ExternalProject)
+project(Boost)
+set(BOOST_VERSION 1.74.0)
+
+list(APPEND BOOST_LIBRARIES
+    "assert"
+    "config"
+    "container_hash"
+    "core"
+    "detail"
+    "dynamic_bitset"
+    "functional"
+    "integer"
+    "move"
+    "preprocessor"
+    "static_assert"
+    "type_traits"
+    "throw_exception"
+)
```

Comment:
Maybe we need Boost.Test here?

Reply:
We in fact do! I'll get that fixed

Reply:
Looks like boost/test depends on quite a lot of other boost libraries. I'm adding all those right now, buts its a painful process. In total I have added 13 new boost library parts to the list to make boost.test work.
We could consider switching to google/googletest since we are already using google/benchmark.